### PR TITLE
Make `Transformation`s classes chainable to form `TransformationList`s

### PR DESCRIPTION
**Pull request by ftynse**
This removes some boilerplate in construction of singleton
transformation lists for the purpose of further composition.

### CODE REVIEW

File: python/matmul/test.py [hunk 1]
```diff
@@ -42,14 +42,14 @@ def __init__(self, tiling_transforms):
...
 # 1 level of tiling and then generalize and interchange.
-expert_tile_1_and_generalize_interchange = Tiling.then(Generalization).then(
-    Vectorization).then(LoweringOnlyExpert)('matmul_on_tensors',
-                                            'linalg.generic',
-                                            tile_sizes=[8, 8, 24],
-                                            tile_interchange=[2, 0, 1],
-                                            pad=False,
-                                            peel=[],
-                                            iterator_interchange=[0, 2, 1])
+expert_tile_1_and_generalize_interchange = Tile.then(Generalize).then(
```
Comment (nicolasvasilache) [inline python/matmul/test.py:45]: Nite: can we make this resemble:

```
expert_tile_1_and_generalize_interchange = \
  Tile('matmul_on_tensors',                              \
         'linalg.generic',                                          \
         tile_sizes=[8, 8, 24],                                 \
         tile_interchange=[2, 0, 1],                        \
         pad=False,                                                 \
         peel=[],                                                      \
         iterator_interchange=[0, 2, 1])                \
  .then(Generalize)                                             \
  .then(Vectorize)                                               \
  .then(LoweringOnlyExpert)
```

or is python going to complain?

Comment (nicolasvasilache) [reply python/matmul/test.py:45]: yes, I insist on the backslashes being at exactly those positions :) ... sigh webUI

**Review by nicolasvasilache — APPROVED** [review]

File: python/matmul/test.py (hunk 1 above)
Comment (ftynse) [reply python/matmul/test.py]: We can't (yet) have parameterized transforms followed by unparameterized transforms, so no options directly to the first `Tile`. Added the backslashes though, I would have hoped for the formatter to put them in.

Comment (nicolasvasilache) [reply python/matmul/test.py]: sg, thanks!

**Pull request merged**